deepchem/models/torch_models/progressive_multitask.py

```diff
@@ -252,14 +261,26 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             out_layer_idx = len(self.layer_sizes)
             x_ = self.layers[task][out_layer_idx](x_)
             if task > 0:
-                adapter_out = self._get_lateral_contrib(logits, task,
+                adapter_out = self._get_lateral_contrib(layer_logits, task,
                                                         out_layer_idx)
                 x_ = x_ + adapter_out
 
-            logits.append(layer_outputs)
-            outputs.append(x_)
-
-        return torch.stack(outputs, dim=1)
+            layer_logits.append(layer_outputs)
+            task_outputs.append(x_)
+
+        output = torch.stack(task_outputs, dim=1)
+
+        if self.mode == 'classification':
```

Comment:
Agreed on separate unit tests for regression/classification

Reply:
Added it